### PR TITLE
Close ZMQ socket after shutdown; remove unnecessary SystemExit

### PR DESCRIPTION
**Pull request by jkrocil**

### CODE REVIEW

File: riggerlib/client.py [hunk 1]
```diff
@@ -115,6 +115,7 @@ def terminate(self):
         try:
             self._request({'event_name': 'shutdown'})
             self.ready = False
+            self.zmq.close()
```
Comment (RonnyPfannschmidt) [inline riggerlib/client.py]: this closes the socket on only one thread, other threads will be unaffected
i do wonder why this works

Comment (jkrocil) [inline riggerlib/client.py]: How many threads with client do we have running at one time?

Comment (jkrocil) [inline riggerlib/client.py]: From what I gather from debugging this, it was only the MainThread.

Comment (jkrocil) [inline riggerlib/client.py]: And even if we had mutliple threads with multiple clients, each would have their own socket open that would be closed here... not sure what you mean by "other threads", please elaborate

Comment (RonnyPfannschmidt) [inline riggerlib/client.py]: we occasionally have a logging call from one of the keepalive threads for the browser/sprout
those thread calls where identified as most probable reason for the jenkins master hangups

Comment (jkrocil) [inline riggerlib/client.py]: But... I assume they instantiate `RiggerClient` or a subclass of it and they also call `terminate` at some point. Correct?

Comment (jkrocil) [inline riggerlib/client.py]: Ooooh... right. They won't call terminate. Got it

Comment (RonnyPfannschmidt) [inline riggerlib/client.py]: the zmq socket is opened as a thread local

so only one of them would be closed in case another thread opens as well

**Pull request closed without merging**